### PR TITLE
using excel-generated test data

### PR DESCRIPTION
**Pull request by ColinEberhardt**
I'm not keen on the current tests, they are quite variable in terms of what they test and their coverage. I  don't think there is much value in testing boundary or default values - these can be tested by visual inspection. I think the most important thing is to test that the algorithm and its implementation are correct!

What I am proposing is the use of an excel spreadsheet with the algorithms implemented, this allows for a very easy verification of their correctness.

The test spec simply reads the data (currently exported manually as CSV), and verifies against our implementation.

How does this sound?

### CODE REVIEW

File: test/bollSpec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        ])
+        .then(result => {
+            const input = result[0];
+            const expectedOutput = result[1];
+
+            // compute the bolinger bands on the Open price
+            const bollingerBands = _bollingerBands();
+            const output = bollingerBands(input.map(d => d.Open));
+
+            expect(output.map(d => d.upper))
```
Comment (ColinEberhardt) [inline test/bollSpec.js]: I also need to add a verification for the lower band and average lines!

**Comment by ColinEberhardt** [comment]
I hope everyone is 👍 to this idea, as I've just finished the last indicator - with a bit of tidy this PR will be ready for review / merge.

It gave me the opportunity to review the existing indicators, which is a good thing :-)

**Comment by ColinEberhardt** [comment]
OK, all updated with tests added for every indicator (apart from percent changed as per #8)

File: src/stochasticOscillator.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default function() {
     };
     stochastic.lowValue = (...args) => {
         if (!args.length) {
-            return highValue;
+            return lowValue;
```
Comment (tlsim) [inline src/stochasticOscillator.js]: Well spotted, thanks.

**Comment by tlsim** [comment]
Looks good to me :+1:

**Comment by ColinEberhardt** [comment]
Thanks @tlsim - I'll land this one ...

**Comment by tlsim** [comment]
@ColinEberhardt It looks like the squashed commit is missing some of the updated *spec.js files. Can you add them?

**Pull request merged**